src/main/java/de/hpi/bpt/argos/api/eventQuery/EventQueryEndpointImpl.java

```diff
@@ -0,0 +1,198 @@
+package de.hpi.bpt.argos.api.eventQuery;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import de.hpi.bpt.argos.api.eventType.EventTypeEndpoint;
+import de.hpi.bpt.argos.api.eventType.EventTypeEndpointImpl;
+import de.hpi.bpt.argos.common.EventPlatformFeedback;
+import de.hpi.bpt.argos.common.EventProcessingPlatformUpdaterImpl;
+import de.hpi.bpt.argos.storage.PersistenceAdapterImpl;
+import de.hpi.bpt.argos.storage.dataModel.event.query.EventQuery;
+import de.hpi.bpt.argos.storage.dataModel.event.query.EventQueryImpl;
+import de.hpi.bpt.argos.util.HttpStatusCodes;
+import de.hpi.bpt.argos.util.LoggerUtilImpl;
+import de.hpi.bpt.argos.util.RestEndpointUtil;
+import de.hpi.bpt.argos.util.RestEndpointUtilImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import static spark.Spark.halt;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EventQueryEndpointImpl  implements EventQueryEndpoint {
+
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeEndpointImpl.class);
+    private static final RestEndpointUtil endpointUtil = RestEndpointUtilImpl.getInstance();
+    private static final JsonParser jsonParser = new JsonParser();
+
+    private static final String JSON_EVENT_QUERY_ATTRIBUTE = "EventQuery";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.post(EventQueryEndpoint.getCreateEventQueryBaseUri(), this::createEventQuery);
+        sparkService.delete(EventQueryEndpoint.getDeleteEventQueryBaseUri(), this::deleteEventQuery);
+        sparkService.put(EventQueryEndpoint.getEditEventQueryBaseUri(), this::editEventQuery);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        try {
+            JsonObject jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+            JsonObject jsonEventQuery = jsonBody.get(JSON_EVENT_QUERY_ATTRIBUTE).getAsJsonObject();
+
+            if (jsonEventQuery == null) {
+                halt(HttpStatusCodes.BAD_REQUEST, "no event query given in body");
+            }
+
+            EventQuery eventQuery = null;
+            try {
+                eventQuery = new EventQueryImpl();
+                eventQuery.setTypeId(jsonEventQuery.get("EventTypeId").getAsLong());

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'jsonEventQuery' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)